app/src/split/central.c

```diff
@@ -0,0 +1,138 @@
+
+#include <errno.h>
+
+#include <zmk/stdlib.h>
+#include <zmk/split/transport/central.h>
+#include <zmk/split/central.h>
+#include <zmk/hid_indicators_types.h>
+#include <zmk/pointing/input_split.h>
+
+#include <zephyr/logging/log.h>
+
+#include <zmk/event_manager.h>
+#include <zmk/events/position_state_changed.h>
+#include <zmk/events/sensor_event.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+// TODO: Active transport selection
+
+struct zmk_split_transport_central *active_transport;
+
+int zmk_split_transport_central_peripheral_event_handler(
+    const struct zmk_split_transport_central *transport, uint8_t source,
+    struct zmk_split_transport_peripheral_event ev) {
+    if (transport != active_transport) {
+        // Ignoring events from non-active transport
+        LOG_WRN("Ignoring peripheral event from non-active transport");
+        return -EINVAL;
+    }
+    switch (ev.type) {
+    case ZMK_SPLIT_TRANSPORT_PERIPHERAL_EVENT_TYPE_KEY_POSITION: {
+        struct zmk_position_state_changed state_ev = {.source = source,
+                                                      .position = ev.data.key_position.position,
+                                                      .state = ev.data.key_position.pressed,
+                                                      .timestamp = k_uptime_get()};
+        return raise_zmk_position_state_changed(state_ev);
+    }
+#if IS_ENABLED(CONFIG_ZMK_INPUT_SPLIT)
+    case ZMK_SPLIT_TRANSPORT_PERIPHERAL_EVENT_TYPE_INPUT_EVENT: {
+        return zmk_input_split_report_peripheral_event(
+            ev.data.input_event.reg, ev.data.input_event.type, ev.data.input_event.code,
+            ev.data.input_event.value, ev.data.input_event.sync);
+    }
+#endif // IS_ENABLED(CONFIG_ZMK_POINTING)
+    case ZMK_SPLIT_TRANSPORT_PERIPHERAL_EVENT_TYPE_SENSOR_EVENT: {
+        struct zmk_sensor_event sensor_ev = {.sensor_index = ev.data.sensor_event.sensor_index,
+                                             .channel_data_size = 1,
+                                             .timestamp = ev.data.sensor_event.timestamp};
+
+        sensor_ev.channel_data[0] = ev.data.sensor_event.channel_data;
+
+        return raise_zmk_sensor_event(sensor_ev);
+    }
+    default:
+        LOG_WRN("GOT AN UNKNOWN EVENT TYPE %d", ev.type);
+        return -ENOTSUP;
+    }
+}
+
+int zmk_split_central_invoke_behavior(uint8_t source, struct zmk_behavior_binding *binding,
```

Comment:
```suggestion
int zmk_split_central_invoke_behavior(uint8_t target, struct zmk_behavior_binding *binding,
```
Minor suggestion, but I just remembered thinking the other day that this name for this variable (and similar variables in functions around the transport) would be nicer. Feel free to ignore and hit resolve.

Reply:
I'm on the fence on this one... It requires just a *tad* bit more mental load to understand source/target are the same thing, just from different perspectives, but I can see where you are coming from. Leaving unresolved for now to sleep on it. Thanks.